docs/guides/using-the-network-shim.md

```diff
@@ -0,0 +1,288 @@
+
+By recording fixtures during a _capture run_ and then using these fixtures during a _stub run_ a full cypress test suite can be executed without a live backend. This is especially useful when running cypress tests on CI. Since it is no longer necessary to have a network connection and a backend, the cypress tests on CI can be more stable. One caveat is that the tests are slightly less realistic, since the network traffic is mocked, so the developer will have to make sure that the recorded fixtures are kept up-to-date.
+
+A capture or stub run is executed for a specific DHIS2 api version and individual fixtures for each version are recorded/played-back during a run. This makes it easier to do cross-version testing for _feature-toggling_ apps.
+
+## Quick setup
+
+The command-line-tool in the package `@dhis2/cli-utils-cypress` comes with an `install` command which generates most of the configuration files documented below. See the [Setup Cypress for your project](https://cli-utils-cypress.dhis2.nu/#/guides/setting-up-cli-tool) guide for more info on how to do this.
+
+To get up-and-running quickly do the following:
+
+1.  Install the cli tool
+
+    ```bash
+    yarn add --dev @dhis2/cli-utils-cypress
+    ```
+
+    <br/>
+
+1.  Issue the install command to generate a working setup
+
+    ```bash
+    d2-utils-cypress install all
+    ```
+
+    (_Instead of `all` another "group" can be specified._)
+    <br/>
+
+1.  Ensure to git-ignore the `./cypress.env.json` file
+    <br/>
+
+1.  Install the peer dependency
+
+    ```bash
+    yarn add cypress
+    ```
+
+    (_Also install `cypress-cucumber` when applicable_)
+    <br/>
+
+1.  Install tools needed to run the scripts in `package.json`:
+    _(or install the `start-server-and-test` package instead)_
+
+    ```bash
+    yarn install -D wait-on concurrently
+    ```
+
+        <br/>
+
+1.  Create scripts `package.json` as demonstrated [here](#package-scripts-and-dependencies)
+    <br/>
+
+1.  Perform a capture run to generate fixtures and commit the generated files:
+
+    ```bash
+    yarn cy:capture
+    ```
+
+    <br/>
+
+1.  Copy-paste GitHub Actions job from [here](https://github.com/dhis2/workflows/blob/master/ci/dhis2-verify-app.yml#L73-L105) into the app's workflow to enable running e2e tests on CI
+    <br/>
+
+## General workflow
+
+Once Network Shim is up-and-running, the general steps required get a cypress test-suite running on CI are as follows:
+
+1. Ensure a live backend is up and running
+1. Ensure a fully passing cypress test-suite is in place
+1. Ensure the CI workflow includes a job that runs the cypress test suite in stub mode
+1. Execute a capture run, which will produce fixture files
+1. Commit the generated fixture files and push to the remote
+1. The test suite will now execute on CI and if a change breaks a test it will be impossible to merge
```

Comment:
I think I'd remove this: `and if a change breaks a test it will be impossible to merge`. Seems a little confusing to me, and the end goal is to get the suite to run on CI, so I think it's fine to stop there. Plus they might not block merging for failed tests, depends on their setup.